### DISCUSSION TITLE
Running composer install shows Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.

### DISCUSSION THREAD

**Discussion by rahuls-gfg** [composer/composer#11407] [Q&A / Support]
Hello Composer and PHP community, 

Our company acquired a new product which is completely built on PHP ecosystem. All of my team are Java and JS devs and we don't have any knowledge of PHP. We decided to learn it. But as of now, when I am trying to run the project in my local machine it throws ```Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.``` error. 

The product was developed in Ukraine and when the war started, the devs have left the development. We can no longer contact them. We neither have any docs for this product too. 

Running ```composer install``` shows the error: 
```
./composer.phar install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
- Required package "guzzlehttp/guzzle" is in the lock file as "5.3.4" but that does not satisfy your constraint "^7.0".
This usually happens when composer files are incorrectly merged or the composer.json file is manually edited.
Read more about correctly resolving merge conflicts https://getcomposer.org/doc/articles/resolving-merge-conflicts.md
and prefer using the "require" command over editing the composer.json file directly https://getcomposer.org/doc/03-cli.md#require-r
Can anyone please help in resolving this issue? 
```

Please find composer.json and composer.lock files below. 

Thank you!
[composer.lock.txt](https://github.com/composer/composer/files/11110476/composer.lock.txt)
[composer.txt](https://github.com/composer/composer/files/11110477/composer.txt)

**Comment by stof** [comment]
Let me quote again the warning you got:

> Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.

The solution for that is given in the warning (see the last sentence).
note that in your case, it should be `./composer.phar update` instead of `composer update` as you haven't installed composer as a global executable.

**Comment by localheinz** [comment]
@rahuls-gfg 

Feel free to get in touch if you need any assistance with your ongoing development!

**Comment by SAULTC001** [comment]
use the next command  composer install --ignore-platform-req=ext-grpc --ignore-platform-req=ext-sodium --ignore-platform-req=ext-gd
